### DISCUSSION TITLE
Auto-deletion of completed tasks??

### DISCUSSION THREAD

**Discussion by vertigo220** [streetwriters/notesnook#1816] [Q&A]
Does this autodelete tasks that are checked off? I had a document I was using for testing, and I had a fairly long checklist with a bunch of items checked off, simulating how I'd use it in practice, checking and unchecking items as needed, and I came back to it a day or two later to find many of the items were gone, and I think they were the ones I had crossed off. If that's really the behavior of this app, it won't work for me, but I want to know if it is or if something else went wrong.

**Comment by thecodrr** [comment]
Hey, nothing ever gets deleted automatically in Notesnook. You must have accidentally pressed the `Clear completed tasks` button which is present at the top bar over each task list. Try checking the session history so you can recover those deleted tasks.

**Reply by vertigo220** [reply]
Thanks. I don't recall doing that, except when testing issue #1793, but I thought I put it all back afterwards. So it's possible I forgot, but I'm keeping an eye on it just in case there was a fluke or bug that caused it to happen. It's good to know it at least isn't supposed to happen, though. I also didn't realize it keeps a history for each session, which would have come in handy and is a really nice feature I'll have to keep in mind in the future.
